client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -21,153 +21,297 @@ import SetupHeader from './setup-header';
-				const progress = this.state.progress + ( 100 / requiredPlugins.length );
-				this.setState( () => ( {
-					installingPlugin: slug,
-					progress,
-				} ) );
-				this.props.installPlugin( site.ID, wporgPlugin );
-				this.setUpdateTimeout();
+	doInstallation = () => {
+		const { site, sitePlugins, translate, wporg } = this.props;
+		const requiredPlugins = this.getRequiredPluginsList();
+
+		// If we are working on nothing presently, get the next thing to install and install it
+		if ( 0 === this.state.workingOn.length ) {
+			const toInstall = this.state.toInstall;
+
+			// Nothing left to install? Advance to activation step
+			if ( 0 === toInstall.length ) {
+				this.setState( {
+					engineState: 'ACTIVATING',
+					message: '',
+					progress: 50,
+				} );
 				return;
 			}
-			if ( ! plugin.active ) {
-				const wporgPlugin = getPlugin( wporg, slug );
-				this.setState( () => ( { activatingPlugin: slug } ) );
-				this.props.activatePlugin( site.ID, { ...wporgPlugin, id: plugin.id } );
-				this.setUpdateTimeout();
+
+			const workingOn = toInstall.shift();
+			if ( 'wc-api-dev' === workingOn ) {
+				// Special handling for wc-api-dev
+				wp.req.post( {
+					path: `/sites/${ site.ID }/woocommerce/install-api-dev-plugin`
+				} ).then( () => {
+					this.props.fetchPlugins( [ site.ID ] );
+				} );
+			} else {
+				// Otherwise, handle plugin installation the normal way
+				this.props.installPlugin( site.ID, getPlugin( wporg, workingOn ) );
+			}
+
+			this.setState( {
+				message: translate( 'Installing %(plugin)s', { args: { plugin: requiredPlugins[ workingOn ] } } ),
+				toInstall,
+				workingOn,
+			} );
+			return;
+		}
+
+		// Otherwise, if we are working on something presently, see if it has appeared in state yet
+		const pluginFound = find( sitePlugins, { slug: this.state.workingOn } );
+		if ( pluginFound ) {
+			this.setState( {
+				workingOn: '',
+			} );
+		}
+	}
+
+	doActivation = () => {
+		const { site, sitePlugins, translate } = this.props;
+		const requiredPlugins = this.getRequiredPluginsList();
+
+		// If we are working on nothing presently, get the next thing to activate and activate it
+		if ( 0 === this.state.workingOn.length ) {
+			const toActivate = this.state.toActivate;
+
+			// Nothing left to activate? Advance to done success
+			if ( 0 === toActivate.length ) {
+				this.setState( {
+					engineState: 'DONESUCCESS',
+					message: '',
+					progress: 100,
+				} );
 				return;
 			}
+
+			const workingOn = toActivate.shift();
+
+			// It is best to use sitePlugins to get the right id since the
+			// plugin id isn't always slug/slug unless the main plugin PHP
+			// file is the same name as the plugin folder
+			const pluginToActivate = find( sitePlugins, { slug: workingOn } );
+			this.props.activatePlugin( site.ID, pluginToActivate );
+
+			this.setState( {
+				message: translate( 'Activating %(plugin)s', { args: { plugin: requiredPlugins[ workingOn ] } } ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Activating %(plugin)s…' )` *ES Score: 20*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).